src/layers/mod.rs

```diff
@@ -102,7 +102,7 @@ impl LayerData {
                     attrs,
                     Some(tilesets),
                     for_tileset,
-                    map_path,
+                    map_path.parent().ok_or(crate::Error::PathIsNotFile)?,
```

Comment:
Could we pass in the parent directory to `LayerData::new` in the first place? Or does the `map_path` parameter to `ImageLayerData::new` and `GroupLayerData::new` need to be the actual map path?

Reply:
We could change that, but I was trying to change as few lines as possible since you asked to patch just the bug.

Reply:
Yeah, that's appreciated! It just highlighted that the inconsistency goes further than this one line. But, we can merge this fix and make it consistent in a follow-up change.

For following this up, I think we should always pass just the directory.